src/sql-parser/src/ast/defs/ddl.rs

```diff
@@ -354,7 +354,14 @@ impl AstDisplay for ColumnDef {
     fn fmt(&self, f: &mut AstFormatter) {
         f.write_node(&self.name);
         f.write_str(" ");
-        f.write_node(&self.data_type);
+        let mut list_depth = 0;
+        let mut dt = &self.data_type;
+        while let DataType::List(t) = dt {
+            dt = t;
+            list_depth += 1;
+        }
+        f.write_node(dt);
+        f.write_str(" list".repeat(list_depth));
```

Comment:
The implementation of `AstDisplay` for `DataType::List` looks wrong. I think that should just read

```rust
            DataType::List(ty) => {
                f.write_node(&ty);
                f.write_str(" list");
            }
```

and then you don't need to hack around that bug here. Unless there's something I'm missing.